Documentation/cross-building.md

```diff
@@ -52,54 +52,103 @@ As usual the resulting binaries will be found in `bin/Product/BuildOS.BuildArch.
 
 Using CoreRT for cross compiling under arm on x86 host
 -----------------------
-It is possible to use CoreRT for compiling under arm/armel on x86 host (or on x64 machine using roots).
+It's possible to use CoreRT for compiling under arm/armel on x86 host (or on x64 machine using rootfs).
+You can build Debug or Release version.
+For example Release means: release CoreRT/CoreCLR/CoreFX builds + CoreRT(ILC) release enabled optimizations.
+For better components compatibility, if you want to build Debug version, you must compile ALL projects as Debug version.
+Otherwise, ALL as Release version.
 
-1. Build CoreCLR for x86 (`checked` version)
+1. Build CoreCLR for x86
 ```
 sudo ./cross/build-rootfs.sh x86 xenial
-./build.sh clang3.9 x86 checked verbose cross skiptests
+./build.sh x86 debug verbose cross
 ```
-
-2. Build CoreFX (`Debug` version)
-3. Build CoreRT for armel, x64, x86
+2. Build CoreFX
+```
+sudo ./cross/build-rootfs.sh x86 xenial
+./build-native.sh -debug -buildArch=x86 -- verbose cross
+./build-managed.sh -debug -verbose
+```
+3. Build CoreRT for x86 and armel
 ```
 sudo ./cross/build-rootfs.sh armel tizen cross
-./build.sh clang3.9 armel debug verbose cross
-./build.sh debug verbose skiptests
-./build.sh clang3.9 x86 debug verbose cross skiptests
+./build.sh armel debug verbose cross
+./build.sh x86 debug verbose cross crosstarget skiptests
 ```
 
-4. Copy necessary binaries to working directory (in x86 rootfs)
+4. Copy necessary binaries to working directory in x86 rootfs.
+Or in any host directory, if you have 32-bit multiarch-support on your x64 host.
 ```
-cp ${CORECLR}/bin/Product/Linux.x86.Checked ${WORKING_DIR}
+# 1) Copy CoreCLR(with CoreRun) part
+cp ${CORECLR}/bin/Product/Linux.x86.Debug/* ${WORKING_DIR}
+
+# 2) Copy CoreRT part
 cp ${CORERT}/bin/Linux.x86.Debug/tools/ilc.dll ${WORKING_DIR}
 cp ${CORERT}/bin/Linux.x86.Debug/tools/ILCompiler.* ${WORKING_DIR}
 cp ${CORERT}/bin/Linux.x86.Debug/tools/System.CommandLine.dll ${WORKING_DIR}
 cp ${CORERT}/bin/Linux.x86.Debug/tools/Microsoft.DiaSymReader.dll ${WORKING_DIR}
 cp ${CORERT}/bin/Linux.x86.Debug/tools/jitinterface.so ${WORKING_DIR}
-cp -r ${CORERT}/bin/Linux.x86.Debug/framework ${WORKING_DIR}
 
-# Copy CoreRT sdk binaries from target (armel) output folder
+# 3) Copy CoreRT framework and sdk binaries from target (armel) output folder
+cp -r ${CORERT}/bin/Linux.armel.Debug/framework ${WORKING_DIR}
 cp -r ${CORERT}/bin/Linux.armel.Debug/sdk ${WORKING_DIR}
+
+# 4) Copy CoreFX part
+# Copy native architecture dependence libs
+cp ${COREFX}/bin/Linux.x86.Debug/native/* ${WORKING_DIR}
+# Copy arch independence libs
+# This part varies depending on the application, these dependencies for HelloWorld only
+NETCORE_PATH=netcoreapp-Linux-Debug-x64
+cp ${COREFX}/bin/runtime/${NETCORE_PATH}/System.Runtime.dll ${WORKING_DIR}
```

Comment:
It looks like you are copying both the compiler (bits that run on x86) and the runtime+framework (bits that run on arm) into the same directory. I think you should keep them separate. Number of .dlls need to be different between the two.